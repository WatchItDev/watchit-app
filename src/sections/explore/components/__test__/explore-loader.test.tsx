import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import ExploreLoader from "../explore-loader";

vi.mock("@src/components/loading-screen", () => ({
  LoadingScreen: () => (
    <div
      data-testid="loading-screen"
      style={{
        zIndex: "999",
        background: "#1E1F22",
        position: "absolute",
        top: "0",
        left: "0",
        bottom: "0",
        pointerEvents: "none",
        right: "0",
        display: "flex",
        justifyContent: "flex-start",
        alignItems: "flex-start",
      }}>
      Loading...
    </div>
  ),
}));

describe("Testing in the <explore-loader/> component ", () => {
  it("should match snapshot", () => {
    const { container } = render(<ExploreLoader />);
    expect(container).toMatchSnapshot();
  });

  it("renders the LoadingScreen component within the layout", () => {
    render(<ExploreLoader />);
    expect(screen.getByTestId("loading-screen")).toBeInTheDocument();
  });

  it("renders the LoadingScreen component with correct styles", () => {
    render(<ExploreLoader />);
    const loadingScreen = screen.getByTestId("loading-screen");
    expect(loadingScreen).toHaveStyle({
      zIndex: "999",
      background: "#1E1F22",
      position: "absolute",
      top: "0",
      left: "0",
      bottom: "0",
      pointerEvents: "none",
      right: "0",
      display: "flex",
      justifyContent: "flex-start",
      alignItems: "flex-start",
    });
  });
});
