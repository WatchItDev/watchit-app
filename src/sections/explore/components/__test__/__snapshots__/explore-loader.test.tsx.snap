// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <explore-loader/> component  > should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1r5plea"
  >
    <div
      class="MuiBox-root css-13nz920"
    >
      <div
        data-testid="loading-screen"
        style="z-index: 999; background: rgb(30, 31, 34); position: absolute; top: 0px; left: 0px; bottom: 0px; pointer-events: none; right: 0px; display: flex; justify-content: flex-start; align-items: flex-start;"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
